### PR TITLE
Fix `getRouteHash` concurrency safety

### PR DESCRIPTION
**Pull request by wnxd**

### CODE REVIEW

**Comment by wnxd** [comment]
Since there is no lock for map reading and writing, panic seems to occur under concurrency.

**Review by TopiSenpai — CHANGES_REQUESTED** [review]
I think I'd be happier to not use `sync.Map` with casting
instead use a `sync.Mutex`

**Comment by wnxd** [comment]
> I think I'd be happier to not use `sync.Map` with casting instead use a `sync.Mutex`

Emmmm, I changed it to `sync.Mutex`.

File: rest/rest_rate_limiter_impl.go [hunk 1]
```diff
@@ -96,15 +97,18 @@ func (l *rateLimiterImpl) Reset() {
 	l.bucketsMu = sync.Mutex{}
 	l.global = 0
 	l.hashes = map[*route.APIRoute]routeHash{}
+	l.hashesMu = sync.Mutex{}
```
Comment (TopiSenpai) [inline rest/rest_rate_limiter_impl.go:100]:
```suggestion
```
zero value mutex is already valid

**Review by TopiSenpai — CHANGES_REQUESTED** [review]
besides that small thingy looks good to me

**Comment by wnxd** [comment]
> besides that small thingy looks good to me

I saw that `l.bucketsMu = sync.Mutex{}` was used above. So I added `l.hashes = sync.Mutex{}`

**Comment by TopiSenpai** [comment]
ohh nevermind it should be there yes

I thought this was in the constructor lol

**Comment by wnxd** [comment]
Ok. I add it.

**Review by TopiSenpai — APPROVED** [review]

**Comment by TopiSenpai** [comment]
thank you!

**Pull request merged**